caddyhttp/proxy/reverseproxy.go

```diff
@@ -189,7 +189,7 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, outreq *http.Request,
 		res.Body.Close()
 		hj, ok := rw.(http.Hijacker)
 		if !ok {
-			return nil
+			panic("not a hijacker")
```

Comment:
Hmm, maybe we should just return an error here instead. And if we can be a little more specific with the message, that would be good: describe what specifically is not a hijacker.